tsm1 checkpoint recovery error

While running the stress tool, the server hung indefinitely on this lock:

https://github.com/influxdb/influxdb/blob/master/tsdb/engine/tsm1/tsm1.go#L2003

I killed the server via `SIGQUIT` and then started it back up and it's unable to start because of this panic:

```
panic: error removing checkpoint file: remove /Users/benbjohnson/.influxdb/data/stress/default/2/0000046.tsm1.check: no such file or directory

goroutine 1 [running]:
github.com/influxdb/influxdb/tsdb/engine/tsm1.(*compactionCheckpoint).cleanup(0xc82025bf80)
	/Users/benbjohnson/go/src/github.com/influxdb/influxdb/tsdb/engine/tsm1/tsm1.go:2131 +0x7a1
github.com/influxdb/influxdb/tsdb/engine/tsm1.(*Engine).cleanupUnfinishedCompaction(0xc820229440)
	/Users/benbjohnson/go/src/github.com/influxdb/influxdb/tsdb/engine/tsm1/tsm1.go:299 +0x8ea
github.com/influxdb/influxdb/tsdb/engine/tsm1.(*Engine).Open(0xc820229440, 0x0, 0x0)
	/Users/benbjohnson/go/src/github.com/influxdb/influxdb/tsdb/engine/tsm1/tsm1.go:227 +0x148
github.com/influxdb/influxdb/tsdb.(*Shard).Open.func1(0xc820272c60, 0x0, 0x0)
	/Users/benbjohnson/go/src/github.com/influxdb/influxdb/tsdb/shard.go:123 +0x365
github.com/influxdb/influxdb/tsdb.(*Shard).Open(0xc820272c60, 0x0, 0x0)
	/Users/benbjohnson/go/src/github.com/influxdb/influxdb/tsdb/shard.go:100 +0x2d
github.com/influxdb/influxdb/tsdb.(*Store).loadShards(0xc820158000, 0x0, 0x0)
	/Users/benbjohnson/go/src/github.com/influxdb/influxdb/tsdb/store.go:303 +0xd2c
github.com/influxdb/influxdb/tsdb.(*Store).Open(0xc820158000, 0x0, 0x0)
	/Users/benbjohnson/go/src/github.com/influxdb/influxdb/tsdb/store.go:371 +0x35a
github.com/influxdb/influxdb/cmd/influxd/run.(*Server).Open.func1(0xc820134360, 0x0, 0x0)
	/Users/benbjohnson/go/src/github.com/influxdb/influxdb/cmd/influxd/run/server.go:392 +0xe08
github.com/influxdb/influxdb/cmd/influxd/run.(*Server).Open(0xc820134360, 0x0, 0x0)
	/Users/benbjohnson/go/src/github.com/influxdb/influxdb/cmd/influxd/run/server.go:338 +0x2d
github.com/influxdb/influxdb/cmd/influxd/run.(*Command).Run(0xc82014e000, 0xc820068180, 0x0, 0x0, 0x0, 0x0)
	/Users/benbjohnson/go/src/github.com/influxdb/influxdb/cmd/influxd/run/command.go:121 +0xe27
main.(*Main).Run(0xc820111f00, 0xc820068180, 0x0, 0x0, 0x0, 0x0)
	/Users/benbjohnson/go/src/github.com/influxdb/influxdb/cmd/influxd/main.go:86 +0x6a1
main.main()
	/Users/benbjohnson/go/src/github.com/influxdb/influxdb/cmd/influxd/main.go:46 +0x29b

goroutine 17 [syscall, locked to thread]:
runtime.goexit()
	/usr/local/go/src/runtime/asm_amd64.s:1696 +0x1

goroutine 19 [syscall]:
os/signal.loop()
	/usr/local/go/src/os/signal/signal_unix.go:22 +0x18
created by os/signal.init.1
	/usr/local/go/src/os/signal/signal_unix.go:28 +0x37

goroutine 35 [IO wait]:
net.runtime_pollWait(0xfc1f50, 0x72, 0xc8200680a0)
	/usr/local/go/src/runtime/netpoll.go:157 +0x60
net.(*pollDesc).Wait(0xc82014a290, 0x72, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:73 +0x3a
net.(*pollDesc).WaitRead(0xc82014a290, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:78 +0x36
net.(*netFD).accept(0xc82014a230, 0x0, 0xf40028, 0xc820010240)
	/usr/local/go/src/net/fd_unix.go:408 +0x27c
net.(*TCPListener).AcceptTCP(0xc820126088, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/tcpsock_posix.go:254 +0x4d
net.(*TCPListener).Accept(0xc820126088, 0x0, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/tcpsock_posix.go:264 +0x3d
github.com/influxdb/influxdb/tcp.(*Mux).Serve(0xc820122fc0, 0xfc0fd0, 0xc820126088, 0x0, 0x0)
	/Users/benbjohnson/go/src/github.com/influxdb/influxdb/tcp/mux.go:48 +0x4b
created by github.com/influxdb/influxdb/cmd/influxd/run.(*Server).Open.func1
	/Users/benbjohnson/go/src/github.com/influxdb/influxdb/cmd/influxd/run/server.go:376 +0xafb

goroutine 36 [chan receive]:
github.com/influxdb/influxdb/tcp.(*listener).Accept(0xc820126090, 0x0, 0x0, 0x0, 0x0)
	/Users/benbjohnson/go/src/github.com/influxdb/influxdb/tcp/mux.go:129 +0x6f
github.com/influxdb/influxdb/meta.(*raftLayer).Accept(0xc82015a9c0, 0x0, 0x0, 0x0, 0x0)
	/Users/benbjohnson/go/src/github.com/influxdb/influxdb/meta/store.go:2129 +0x51
github.com/hashicorp/raft.(*NetworkTransport).listen(0xc82014a310)
	/Users/benbjohnson/go/src/github.com/hashicorp/raft/net_transport.go:346 +0x79
created by github.com/hashicorp/raft.NewNetworkTransport
	/Users/benbjohnson/go/src/github.com/hashicorp/raft/net_transport.go:138 +0x284

goroutine 37 [select]:
github.com/hashicorp/raft.(*Raft).leaderLoop(0xc820178000)
	/Users/benbjohnson/go/src/github.com/hashicorp/raft/raft.go:826 +0xcce
github.com/hashicorp/raft.(*Raft).runLeader(0xc820178000)
	/Users/benbjohnson/go/src/github.com/hashicorp/raft/raft.go:798 +0x88e
github.com/hashicorp/raft.(*Raft).run(0xc820178000)
	/Users/benbjohnson/go/src/github.com/hashicorp/raft/raft.go:595 +0xbb
github.com/hashicorp/raft.(*Raft).(github.com/hashicorp/raft.run)-fm()
	/Users/benbjohnson/go/src/github.com/hashicorp/raft/raft.go:245 +0x20
github.com/hashicorp/raft.(*raftState).goFunc.func1(0xc820178000, 0xc820155be0)
	/Users/benbjohnson/go/src/github.com/hashicorp/raft/state.go:152 +0x44
created by github.com/hashicorp/raft.(*raftState).goFunc
	/Users/benbjohnson/go/src/github.com/hashicorp/raft/state.go:153 +0x4d

goroutine 38 [select]:
github.com/hashicorp/raft.(*Raft).runFSM(0xc820178000)
	/Users/benbjohnson/go/src/github.com/hashicorp/raft/raft.go:502 +0x10eb
github.com/hashicorp/raft.(*Raft).(github.com/hashicorp/raft.runFSM)-fm()
	/Users/benbjohnson/go/src/github.com/hashicorp/raft/raft.go:246 +0x20
github.com/hashicorp/raft.(*raftState).goFunc.func1(0xc820178000, 0xc820155bf0)
	/Users/benbjohnson/go/src/github.com/hashicorp/raft/state.go:152 +0x44
created by github.com/hashicorp/raft.(*raftState).goFunc
	/Users/benbjohnson/go/src/github.com/hashicorp/raft/state.go:153 +0x4d

goroutine 39 [select]:
github.com/hashicorp/raft.(*Raft).runSnapshots(0xc820178000)
	/Users/benbjohnson/go/src/github.com/hashicorp/raft/raft.go:1613 +0x493
github.com/hashicorp/raft.(*Raft).(github.com/hashicorp/raft.runSnapshots)-fm()
	/Users/benbjohnson/go/src/github.com/hashicorp/raft/raft.go:247 +0x20
github.com/hashicorp/raft.(*raftState).goFunc.func1(0xc820178000, 0xc820155c00)
	/Users/benbjohnson/go/src/github.com/hashicorp/raft/state.go:152 +0x44
created by github.com/hashicorp/raft.(*raftState).goFunc
	/Users/benbjohnson/go/src/github.com/hashicorp/raft/state.go:153 +0x4d

goroutine 40 [select]:
github.com/influxdb/influxdb/meta.(*localRaft).logLeaderChanges(0xc82011cc80)
	/Users/benbjohnson/go/src/github.com/influxdb/influxdb/meta/state.go:189 +0x4f5
created by github.com/influxdb/influxdb/meta.(*localRaft).open
	/Users/benbjohnson/go/src/github.com/influxdb/influxdb/meta/state.go:179 +0x1344

goroutine 41 [chan receive]:
github.com/influxdb/influxdb/tcp.(*listener).Accept(0xc820126098, 0x0, 0x0, 0x0, 0x0)
	/Users/benbjohnson/go/src/github.com/influxdb/influxdb/tcp/mux.go:129 +0x6f
github.com/influxdb/influxdb/meta.(*Store).serveExecListener(0xc820158140)
	/Users/benbjohnson/go/src/github.com/influxdb/influxdb/meta/store.go:622 +0xa2
created by github.com/influxdb/influxdb/meta.(*Store).Open
	/Users/benbjohnson/go/src/github.com/influxdb/influxdb/meta/store.go:239 +0x364

goroutine 42 [chan receive]:
github.com/influxdb/influxdb/tcp.(*listener).Accept(0xc8201260a0, 0x0, 0x0, 0x0, 0x0)
	/Users/benbjohnson/go/src/github.com/influxdb/influxdb/tcp/mux.go:129 +0x6f
github.com/influxdb/influxdb/meta.(*Store).serveRPCListener(0xc820158140)
	/Users/benbjohnson/go/src/github.com/influxdb/influxdb/meta/store.go:726 +0xa2
created by github.com/influxdb/influxdb/meta.(*Store).Open
	/Users/benbjohnson/go/src/github.com/influxdb/influxdb/meta/store.go:242 +0x3b6

goroutine 5 [select]:
github.com/influxdb/influxdb/cmd/influxd/run.(*Server).monitorErrorChan(0xc820134360, 0xc82012e4e0)
	/Users/benbjohnson/go/src/github.com/influxdb/influxdb/cmd/influxd/run/server.go:531 +0x145
created by github.com/influxdb/influxdb/cmd/influxd/run.(*Server).Open.func1
	/Users/benbjohnson/go/src/github.com/influxdb/influxdb/cmd/influxd/run/server.go:382 +0xc7f

goroutine 6 [select]:
github.com/influxdb/influxdb/meta.(*Store).WaitForLeader(0xc820158140, 0x6fc23ac00, 0x0, 0x0)
	/Users/benbjohnson/go/src/github.com/influxdb/influxdb/meta/store.go:556 +0x3fc
github.com/influxdb/influxdb/monitor.(*Monitor).storeStatistics(0xc8201480e0)
	/Users/benbjohnson/go/src/github.com/influxdb/influxdb/monitor/service.go:341 +0x2dd
created by github.com/influxdb/influxdb/monitor.(*Monitor).Open
	/Users/benbjohnson/go/src/github.com/influxdb/influxdb/monitor/service.go:138 +0x349
```

Here's the contents of that directory:

```
$ ls -al ~/.influxdb/data/stress/default/2
total 549952
drwxr-xr-x  23 benbjohnson  staff        782 Oct 22 10:48 .
drwx------   4 benbjohnson  staff        136 Oct 22 10:30 ..
-rw-r--r--   1 benbjohnson  staff    4804269 Oct 22 10:41 0000044.tsm1
-rw-r--r--   1 benbjohnson  staff    5754566 Oct 22 10:41 0000045.tsm1
-rw-r--r--   1 benbjohnson  staff        561 Oct 22 10:48 0000046.compact
-rw-r--r--   1 benbjohnson  staff   92472278 Oct 22 10:44 0000046.tsm1
-rw-r--r--   1 benbjohnson  staff    4986479 Oct 22 10:42 0000047.tsm1
-rw-r--r--   1 benbjohnson  staff    4174146 Oct 22 10:42 0000050.tsm1
-rw-r--r--   1 benbjohnson  staff    4138344 Oct 22 10:43 0000053.tsm1
-rw-r--r--   1 benbjohnson  staff    4888084 Oct 22 10:44 0000054.tsm1
-rw-r--r--   1 benbjohnson  staff    5397278 Oct 22 10:45 0000058.tsm1
-rw-r--r--   1 benbjohnson  staff    5141573 Oct 22 10:45 0000060.tsm1
-rw-r--r--   1 benbjohnson  staff  120201530 Oct 22 10:48 0000061.tsm1
-rw-r--r--   1 benbjohnson  staff    3735902 Oct 22 10:45 0000062.tsm1
-rw-r--r--   1 benbjohnson  staff    4920712 Oct 22 10:46 0000063.tsm1
-rw-r--r--   1 benbjohnson  staff    1901348 Oct 22 10:46 0000065.tsm1
-rw-r--r--   1 benbjohnson  staff    4716518 Oct 22 10:46 0000066.tsm1
-rw-r--r--   1 benbjohnson  staff    5528751 Oct 22 10:47 0000068.tsm1
-rw-r--r--   1 benbjohnson  staff    3517831 Oct 22 10:47 0000069.tsm1
-rw-r--r--   1 benbjohnson  staff    4738569 Oct 22 10:47 0000070.tsm1
-rw-r--r--   1 benbjohnson  staff         69 Oct 22 10:30 fields
-rw-r--r--   1 benbjohnson  staff     280430 Oct 22 10:30 ids
-rw-r--r--   1 benbjohnson  staff     227863 Oct 22 10:30 series
```

--- a/tsdb/engine/tsm1/tsm1.go
+++ b/tsdb/engine/tsm1/tsm1.go
@@ -302,7 +302,7 @@ func (e *Engine) cleanupUnfinishedCompaction() {
 		if err := f.Close(); err != nil {
 			panic(fmt.Sprintf("error closing compaction checkpoint: %s", err.Error()))
 		}
-		if err := os.Remove(f.Name()); err != nil {
+		if err := os.RemoveAll(f.Name()); err != nil {
 			panic(fmt.Sprintf("error removing compaction checkpoint: %s", err.Error()))
 		}
 	}
@@ -720,7 +720,7 @@ func (e *Engine) clearCompactedFiles(compactionCheckpointName string, newFiles [
 		}
 
 		// finally remove the compaction marker
-		if err := os.Remove(compactionCheckpointName); err != nil {
+		if err := os.RemoveAll(compactionCheckpointName); err != nil {
 			e.logger.Printf("error removing %s: %s", compactionCheckpointName, err.Error())
 		}
 
@@ -1533,7 +1533,7 @@ func (e *Engine) writeFields(fields map[string]*tsdb.MeasurementFields) error {
 	fieldsFileName := filepath.Join(e.path, FieldsFileExtension)
 
 	if _, err := os.Stat(fieldsFileName); !os.IsNotExist(err) {
-		if err := os.Remove(fieldsFileName); err != nil {
+		if err := os.RemoveAll(fieldsFileName); err != nil {
 			return err
 		}
 	}
@@ -1607,7 +1607,7 @@ func (e *Engine) writeSeries(series map[string]*tsdb.Series) error {
 	seriesFileName := filepath.Join(e.path, SeriesFileExtension)
 
 	if _, err := os.Stat(seriesFileName); !os.IsNotExist(err) {
-		if err := os.Remove(seriesFileName); err != nil && err != os.ErrNotExist {
+		if err := os.RemoveAll(seriesFileName); err != nil {
 			return err
 		}
 	}
@@ -1699,12 +1699,12 @@ func (e *Engine) removeFileIfCheckpointExists(fileName string) bool {
 	}
 
 	// there's a checkpoint so we know this file isn't safe so we should remove it
-	err = os.Remove(fileName)
+	err = os.RemoveAll(fileName)
 	if err != nil {
 		panic(fmt.Sprintf("error removing file %s", err.Error()))
 	}
 
-	err = os.Remove(checkpointName)
+	err = os.RemoveAll(checkpointName)
 	if err != nil {
 		panic(fmt.Sprintf("error removing file %s", err.Error()))
 	}
@@ -2005,7 +2005,7 @@ func (d *dataFile) Delete() error {
 	if err := d.close(); err != nil {
 		return err
 	}
-	err := os.Remove(d.f.Name())
+	err := os.RemoveAll(d.f.Name())
 	if err != nil {
 		return err
 	}
@@ -2127,17 +2127,17 @@ type compactionCheckpoint struct {
 func (c *compactionCheckpoint) cleanup() {
 	for _, fn := range c.CompactedFiles {
 		cn := checkpointFileName(fn)
-		if err := os.Remove(cn); err != nil && err != os.ErrNotExist {
+		if err := os.RemoveAll(cn); err != nil {
 			panic(fmt.Sprintf("error removing checkpoint file: %s", err.Error()))
 		}
-		if err := os.Remove(fn); err != nil && err != os.ErrNotExist {
+		if err := os.RemoveAll(fn); err != nil {
 			panic(fmt.Sprintf("error removing old data file: %s", err.Error()))
 		}
 	}
 
 	for _, fn := range c.NewFiles {
 		cn := checkpointFileName(fn)
-		if err := os.Remove(cn); err != nil && err != os.ErrNotExist {
+		if err := os.RemoveAll(cn); err != nil {
 			panic(fmt.Sprintf("error removing checkpoint file: %s", err.Error()))
 		}
 	}
@@ -2235,7 +2235,7 @@ func checkpointFileName(fileName string) string {
 // removeCheckpoint removes the checkpoint for a new data file that was getting written
 func removeCheckpoint(fileName string) error {
 	checkpointFile := fmt.Sprintf("%s.%s", fileName, CheckpointExtension)
-	return os.Remove(checkpointFile)
+	return os.RemoveAll(checkpointFile)
 }
 
 // u64tob converts a uint64 into an 8-byte slice.

--- a/tsdb/engine/tsm1/tsm1.go
+++ b/tsdb/engine/tsm1/tsm1.go
@@ -1340,9 +1340,6 @@ func (e *Engine) replaceCompressedFile(name string, data []byte) error {
 	if err := f.Close(); err != nil {
 		return err
 	}
-	if err := os.Remove(name); err != nil && !os.IsNotExist(err) {
-		return err
-	}
 	return os.Rename(tmpName, filepath.Join(e.path, name))
 }
 
@@ -1531,13 +1528,6 @@ func (e *Engine) writeFields(fields map[string]*tsdb.MeasurementFields) error {
 		return err
 	}
 	fieldsFileName := filepath.Join(e.path, FieldsFileExtension)
-
-	if _, err := os.Stat(fieldsFileName); !os.IsNotExist(err) {
-		if err := os.RemoveAll(fieldsFileName); err != nil {
-			return err
-		}
-	}
-
 	return os.Rename(fn, fieldsFileName)
 }
 
@@ -1605,13 +1595,6 @@ func (e *Engine) writeSeries(series map[string]*tsdb.Series) error {
 		return err
 	}
 	seriesFileName := filepath.Join(e.path, SeriesFileExtension)
-
-	if _, err := os.Stat(seriesFileName); !os.IsNotExist(err) {
-		if err := os.RemoveAll(seriesFileName); err != nil {
-			return err
-		}
-	}
-
 	return os.Rename(fn, seriesFileName)
 }
 

